src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -554,7 +554,9 @@ struct
     let soon_to_be_root_merkle_root =
       Ledger.merkle_root soon_to_be_root_ledger
     in
+    let%bind () = Async.Scheduler.yield () in
     Ledger.commit soon_to_be_root_ledger ;
+    let%map () = Async.Scheduler.yield () in
```

Comment:
Can you explain why we need this yield in the end here?

Reply:
Commiting the root takes a long time and we want split up the async cycles we use into the smallest chunks possible. This gives the scheduler a chance to schedule other cycles before we continue execution after performing this long task.